libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2196,7 +2196,10 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 	for (size_t i = 0; i < numParameters; i++)
 	{
 		Type const& argType = *type(*callArguments[i]);
-		BoolResult result = argType.isImplicitlyConvertibleTo(*functionPointerType->parameterTypes()[i]);
+		// functionPointerType is never equal to DelegateCall ( it would have thrown error above )
```

Comment:
I'd say you can either just drop the comment or if it should be more explicit make it a proper
```suggestion
		solAssert(functionPointerType->kind() != FunctionType::Kind::Declaration);
```

Reply:
See line 2124. I would actually do the conversion to external callable function much closer to where the pointer is defined.

Reply:
Yeah, I was actually wondering whether to suggest that as well!